src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/InterfaceDispatchMapIndirectionNode.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Internal.TypeSystem;
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    class InterfaceDispatchMapIndirectionNode : EmbeddedObjectNode, ISymbolNode
```

Comment:
Would it be possible to use `ArrayOfEmbeddedPointersNode` with a set of `EmbeddedPointerIndirectionNode`s in place of this?

Feel free to hack on those two classes and provide additional hooks if needed. I know the design could use another pair of eyes to make general purpose. But they're meant to replace boilerplate code like this.

I don't insist on it too much. There are a bunch of other places I'm planning to replace with the new class when time allows. One extra won't kill me.